lib/core/macros.nim

```diff
@@ -1042,6 +1042,55 @@ macro expandMacros*(body: typed): untyped =
   result = getAst(inner(body))
   echo result.toStrLit
 
+proc pragmaNodeForAttrSubj(subj: NimNode): NimNode =
+  var subj = subj
+  if subj.kind == nnkSym:
+    subj = getImpl(subj.symbol)
+    subj.expectRoutine()
+    result = subj.pragma
+  elif subj.kind == nnkDotExpr:
+    let typDef = getImpl(getTypeInst(subj[0]).symbol)
+    typDef.expectKind(nnkTypeDef)
+    typDef[2].expectKind(nnkObjectTy)
+    let recList = typDef[2][2]
+    for identDefs in recList:
+      for i in 0 ..< identDefs.len - 2:
+        if identDefs[i].kind == nnkPragmaExpr and identDefs[i][0].kind == nnkIdent and $identDefs[i][0] == $subj[1]:
```

Comment:
But ``identDefs[i][0].kind == nnkIdent`` is wrong, it could be a dot node. The point was to allow namespacing of annotations.